scripts/release.js

```diff
@@ -0,0 +1,214 @@
+    console.log(
+      colors.blue(`
+      - Current version is "${currentVersion}"
+      - Changelog will be generated only if a fix/feat/performance/breaking is found in git log
+      - You must choose a new ve.rs.ion (semver) ${strategy === 'beta'
+        ? 'with -beta.x suffix'
+        : ''}
+    `)
+    );
+
+    execSync(
+      'conventional-changelog --preset angular --output-unreleased | less'
+    );
+
+    inquirer
+      .prompt([
+        {
+          type: 'input',
+          name: 'newVersion',
+          message: colors.blue.underline(
+            strategy === 'stable'
+              ? 'Please type the new chosen version'
+              : 'Please type the new chosen version (with -beta.x suffix)'
+          ),
+          validate(newVersion) {
+            return (
+              strategy === 'stable' ||
+              (newVersion && /-beta\.\d{1,}$/.test(newVersion)) ||
+              'You must include `-beta.x` suffix in the version number'
+            );
+          },
+        },
+      ])
+      .then(({ newVersion }) => {
+        // bump new version
+        console.log(colors.blue(`Bump version to "${newVersion}"`));
+        execSync(
+          `VERSION=${newVersion} babel-node ./scripts/bump-package-version.js`
+        );
+
+        // build library new version
+        console.log(colors.blue('Building library new version'));
+        execSync(`NODE_ENV=production VERSION=${newVersion} npm run build`);
+
+        // update changelog
+        console.log(colors.blue('Update changelog'));
+        const changelog = execSync('conventional-changelog -p angular')
+          .toString()
+          .trim();
+
+        execSync(
+          'conventional-changelog --preset angular --infile CHANGELOG.md --same-file'
+        );
+
+        // regenerate README TOC
+        console.log(colors.blue('Generate TOCS'));
+        execSync('npm run doctoc');
+
+        // regenerate yarn.lock
+        execSync('yarn');
+
+        // git add and tag
+        const commitMessage = `v${newVersion}\n\n${changelog}`;
+        execSync(
+          'git add src/lib/version.js yarn.lock package.json CHANGELOG.md README.md CONTRIBUTING.md'
+        );
+        execSync(`git commit -m "${commitMessage}"`);
+        execSync(`git tag "v${newVersion}"`);
+
+        console.log(
+          colors.yellow.underline(
+            'Almost done, check everything in another terminal tab'
+          )
+        );
+
+        inquirer
+          .prompt([
+            {
+              type: 'confirm',
+              name: 'publishNpm',
+              message: colors.yellow.underline(
+                'Is everything OK? Continue to push to github and publish the package?'
+              ),
+            },
+          ])
+          .then(({ publishNpm }) => {
+            if (!publishNpm) rollback(newVersion);
+
+            console.log(colors.blue('Push to github, publish on npm'));
+            if (strategy === 'stable') {
```

Comment:
I did some functions before, it was harder to read because going to reference to references. Since it's a script I wanted something straightforward that you can read like a text from the top to the bottom without jumping elsewhere to avoid mistakes when we will work on it again.